src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -2474,6 +2474,61 @@ void Main()
                 );
         }
 
+        [Fact, WorkItem(22662, "https://github.com/dotnet/roslyn/issues/22662")]
+        public void LambdaSquigglesArea()
+        {
+            var src = @"
+class C
+{
+    void M()
+    {
+        System.Func<bool, System.Action<bool>> x = x1 => x2 =>
```

Comment:
>x2 => [](start = 57, length = 5)

It would be good to add tests with ```delegate ...``` expression.

Reply:
Thanks. Added handling for that.